src/fusion_power_plant.cc

```diff
@@ -10,14 +10,98 @@ std::string FusionPowerPlant::str() {
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    //Set up a schedule fill policy with Katie's active/dormant policy code
+    //(not shown here). Do not start the policy yet.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (can_operate) {
+    SequesterTritium();
+    OperateReactor();
+  }
+
+  //I think somehow combining all these would be nice...
+  DecayInventory(tritium_storage);
+  ExtractHelium(tritium_storage);
+
+  DecayInventory(tritium_excess);
+  ExtractHelium(tritium_excess);
+
+  sequestered_tritium->Decay(context()->time());
+
+  //This was a bunch of free-floating code earlier, but i've truncated it 
+  //because it's mostly for DRE stuff.
+  MoveExcessTritiumToSellBuffer();
+
+  //This maybe belongs in its own function?
+  if (context()->time() % blanket_turnover_frequency == 0 && !blanket.empty()) {
+    if (blanket.quantity() >= blanket_turnover) {
+      blanket_excess.Push(blanket.Pop(blanket_turnover));
+      CombineInventory(blanket_excess);
+      RecordOperationalInfo("Blanket Cycled");
+    } else {
+      RecordOperationalInfo("Blanket Not Cycled");
+    }
+  }
+  
+
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tock() {
-  std::cout<<"There!"<<std::endl;
+  //pseudocode implementation of Tock():
+  
+  if(!can_operate) {
+    try { 
+      CheckOperatingConditions(); //Previously "Startup()"
+      fuel_startup_policy.Stop();
+      fuel_refill_policy.Start();
+    } catch (const std::exception& e) {
+      RecordError("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "FusionPowerPlant") << e.what();
+    }
```

Comment:
exceptions shouldn't be the way to handle the set of regularly possible states

Reply:
I think I was doing this for two reasons: I remember there being some sort of simulation-ending error hidden in here that I was getting around with the catch, and two, within that this was a convenient way to let the user know what was going wrong. Definitely worth revisiting, though. I'll change it in the pseudocode implementation and then if we run into problems we can revisit this type of solution (or something better if there is such a thing)